src/App/Pages/Accounts/SetPasswordPageViewModel.cs

```diff
@@ -186,8 +186,10 @@ public async Task SubmitAsync()
                 MasterPasswordHash = masterPasswordHash,
                 Key = encKey.Item2.EncryptedString,
                 MasterPasswordHint = Hint,
-                Kdf = kdf,
-                KdfIterations = Constants.KdfIterations,
+                Kdf = kdfConfig.Type.GetValueOrDefault(KdfType.PBKDF2_SHA256),
+                KdfIterations = kdfConfig.Iterations.GetValueOrDefault(Constants.KdfIterations),
```

Comment:
The `Type` and Iterations are already set above with the default values so no need to add `GetValueOrDefault(...)` just use `kdfConfig.Type.Value` and `kdfConfig.Iterations.Value`

Reply:
This is due to type and iterations being optional in the kdfconfig struct. Changing them to non-optional in the struct breaks some other things (f.e setting default iteration count in the CryptoService in the MakeKeyAsync function.